### ISSUE TITLE
feat: users should be able to customize accessibility defaults

### ISSUE THREAD

**Issue by liamdebeasi** [ionic-team/ionic-framework#21327]
# Bug Report

**Ionic version:**

[x] **5.x**

**Current behavior:**

Often times users will want to customize the accessibility defaults we provide for components. For example, if a user is distributing an app in multiple languages, having a label always be announced in English is not very helpful. Additionally, even if users are distributing an app in English only, they cannot override the defaults we provide.

For example, we add `role="banner"` to the Host of `ion-header` (https://github.com/ionic-team/ionic/blob/master/core/src/components/header/header.tsx#L148). This may not be desirable to some users and they may wish to use `role="none"`; however, they are unable to override this as `role="banner"` is always used. On top of this, if we provide any attributes to elements inside the shadow dom, users cannot easily access them to customize them. Users can access them via JS and `shadowRoot`, but that is not a sustainable approach.

**Expected behavior:**

I would expect that developers can customize most/all of the accessibility defaults we provide to fine tune the user experience for their applications.

This thread will be used to track areas where there needs to be improvement regarding this.

**Comment by liamdebeasi** [comment]
Hi @paulobr01 and @masbrutus,

Thanks for the feedback! Let me look into this issue and see if we can make `role` easier to customize on `ion-header`.

**Comment by liamdebeasi** [comment]
Hey there,

I have merged in https://github.com/ionic-team/ionic-framework/pull/23888 which lets developers customize the `role` attribute on `ion-header`. This addition should be available in the next release of Ionic Framework. Thanks for reporting this!

--------

On a separate note regarding the GitHub issue itself, I am going to close this out in favor of having one GitHub thread per issue as opposed to having all issues tracked in this thread. This makes it much easier for us to track progress and provide updates to the community.

For any additional issues, please feel free to open separate GitHub issues. Thank you!
